plugin/storage/memory/memory.go

```diff
@@ -59,8 +74,32 @@ func WithConfiguration(configuration config.Configuration) *Store {
 	}
 }
 
+// getTenant returns the per-tenant storage.  Note that tenantID has already been checked for by the collector or query
+func (st *Store) getTenant(tenantID string) *Tenant {
+	tenant, ok := st.perTenant[tenantID]
+	if !ok {
+		// We do the lookup twice to skip locking on retrieval of existing tenant
```

Comment:
Can't really do this because you're doing unprotected read above that could cause a concurrency issue. Test with `-race` should catch this. If you're worried about locking, you can either use RWLock or use a sync.Map which is specifically designed for frequent reads / rare writes.